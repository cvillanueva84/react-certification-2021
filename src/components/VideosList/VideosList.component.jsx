import React from 'react';
import styled from 'styled-components';
import { items } from './youtube-videos-mock.json';

const { snippet: channel } = items.shift();

const VideoCard = styled.div`
  max-width: 320px;
  margin: 0.75rem;
`;

const Wrapper = styled.div`
  display: flex;
  align-items: flex-start;
`;

const Link = styled.a`
  color: #000000;
  font-weight: 700;
`;

const ChannelAvatar = styled.img`
  width: 40px;
  height: 40px;
  border-radius: 50%;
`;

const Text = styled.p`
  margin: 0;
  color: #555555;
`;

const MarginX = styled.div`
  margin: 0 0.75rem;
`;

const VideosList = () =>
  items.map(({ id, snippet }) => (
    <VideoCard key={id.videoId}>
      <figure>
        <img src={snippet.thumbnails.medium.url} alt={`${snippet.title} thumbnail`} />
      </figure>
      <Wrapper>
        <ChannelAvatar src={channel.thumbnails.default.url} alt={channel.channelTitle} />
        <MarginX className="margin-x">
          <Link href={`https://www.youtube.com/watch?v=${id.videoId}`}>
            {snippet.title}
          </Link>
          <div>
            <Text>{snippet.channelTitle}</Text>
            <Text>{snippet.publishedAt}</Text>
          </div>
        </MarginX>
      </Wrapper>
    </VideoCard>
  ));

export default VideosList;
