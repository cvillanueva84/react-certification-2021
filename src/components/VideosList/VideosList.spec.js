import VideosList from './VideosList.component';

const { render, screen } = require('@testing-library/react');

// test.js
jest.mock('./youtube-videos-mock.json');

describe('VideosList', () => {
  let container;

  beforeEach(() => {
    container = render(<VideosList />).container;
  });

  test('Should renders VideosList component', () => {
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('Should contains an image with the thumbnail url', () => {
    const videoThumbnailImageElement = screen.getByAltText('fake video title thumbnail');
    expect(videoThumbnailImageElement).toBeInTheDocument();
    expect(videoThumbnailImageElement).toHaveAttribute('src', 'fake-video-url-image.png');
  });
  test('Should contains an image with the channel avatar', () => {
    const channelImageElement = screen.getByAltText('fake channel title');
    expect(channelImageElement).toBeInTheDocument();
    expect(channelImageElement).toHaveAttribute('src', 'fake-channel-image.png');
  });

  test('Should contains a link redirects to the video url', () => {
    const linkELement = screen.getByText(/fake video title/i);
    expect(linkELement).toBeInTheDocument();
    expect(linkELement).toHaveAttribute(
      'href',
      'https://www.youtube.com/watch?v=fakeVideoId'
    );
  });

  test('Should shows the channel title', () => {
    const channelTitleElement = screen.getByText(/fake channel title/i);
    expect(channelTitleElement).toBeInTheDocument();
  });

  test('Should shows the video published date', () => {
    const channelTitleElement = screen.getByText(/2014-09-27T01:39:18Z/i);
    expect(channelTitleElement).toBeInTheDocument();
  });
});
