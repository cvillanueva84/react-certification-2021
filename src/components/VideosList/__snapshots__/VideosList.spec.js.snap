// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideosList Should renders VideosList component 1`] = `
<div>
  <div
    class="sc-bdnxRM iIRjbH"
  >
    <figure>
      <img
        alt="fake video title thumbnail"
        src="fake-video-url-image.png"
      />
    </figure>
    <div
      class="sc-gtsrHT eMBaCB"
    >
      <img
        alt="fake channel title"
        class="sc-hKFxyN bdMQIT"
        src="fake-channel-image.png"
      />
      <div
        class="sc-jSFjdj bgycdW margin-x"
      >
        <a
          class="sc-dlnjwi hzuETT"
          href="https://www.youtube.com/watch?v=fakeVideoId"
        >
          fake video title
        </a>
        <div>
          <p
            class="sc-eCApnc fpRBdY"
          >
            fake channel title
          </p>
          <p
            class="sc-eCApnc fpRBdY"
          >
            2014-09-27T01:39:18Z
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
